### PR TITLE
BedTools & MultiProcessing

### PR DESCRIPTION
**Pull request by fayyazh**
1. Shall we also define the distance as an input instead of fix? 
2. Initial input still kept divided into two processes because its giving better performance.

### CODE REVIEW

File: mean_shift_sklearn.py [hunk 1]
```diff
@@ -167,16 +189,43 @@ def main():
...
+    input_file = BedTool(args.data_file)
+    merged_data = BedTool(input_file.sort().
+                          merge(s=True, c=(2, 6), d=500, o=('max', 'first')))
+    intersected_data = merged_data.\
+        intersect(s=True, loj=True, a=input_file, b=merged_data)
+
     # retrieves chromosome name, start and strand from input bed file
-    bed_file = np.loadtxt(args.data_file, dtype=str, usecols=(0, 1, 5))
+    bed_data = []
+    for data in intersected_data:
```
Comment (dmaticzka) [inline mean_shift_sklearn.py]: please add comments about what data columns you are selecting and why

File: mean_shift_sklearn.py [hunk 2]
```diff
@@ -44,21 +47,24 @@ def group_chrm(self, bed_file):
...
+        [(('chrX', '+', '1'), ['1', '2', '3', '4', '5']),\
+ (('chrX', '-', '9'), ['9', '10', '11', '12', '13']),\
+ (('chrY', '+', '20'), ['20', '21', '22', '23', '24'])]
         """
+
         chrm_details = dict()
 
         for b in bed_file:
-            chrm_name_strand = (b[0], b[2])
+            chrm_name_strand = (b[0], b[2], b[3])
```
Comment (dmaticzka) [inline mean_shift_sklearn.py]: also please annotate what is in the selected columns

Comment (dmaticzka) [reply mean_shift_sklearn.py]: chrm_name_strand is the columns? please name these according to what they are used for, e.g. key? and mention its content via comment

**Review by dmaticzka — COMMENTED** [review]
Neat, thank you! Please comment on the selected columns and we can merge.

**Comment by dmaticzka** [comment]
Having the distance as input parameter with default is a good idea.

Regarding the two processes, can you calculate that via a pool like you do in call_centroid_and_output? Then we can very easily increase the one or the other. What is fastest is likely to change when running on a cluster machine with 24 cores.

**Comment by fejjaz** [comment]
regarding the two processes, can be swapped with sub-class multiprocessing.pool.Pool instead of multiprocessing.Pool. It's only practical to do if we split the dictionary before calling ms.call_centroid_and_output otherwise it would be just one element of a list as a dictionary so splitting into two

**Comment by fayyazh** [comment]
-  instead of Process object, Pool used.
-  '-p' parameter used to define processes/cores to use. default 2 
-  details comments added.

**Comment by dmaticzka** [comment]
Thank you Fejjaz, looks really good!

**Pull request merged**